lib/spack/spack/util/environment.py

```diff
@@ -744,11 +744,11 @@ def from_sourcing_file(
             ]
         )
 
+        # Implements a no op during forwards operation on respective os
+        null_fd = "echo > NUL" if sys.platform == "win32" else os.devnull
```

Comment:
I'm not sure how this fits the API of `environment_after_sourcing_files`: for that function the first argument is supposed to be a file path or a tuple. I can sort of see how this might work by tracing through the logic, but I think it might match the expected API better to pass it an empty file (e.g. created by `tempfile`).

Reply:
I retooled this, let me know what you think